pkg/util/tracing/collector/collector_test: TestClusterInflightTraces failed

pkg/util/tracing/collector/collector_test.TestClusterInflightTraces [failed](https://tanzanite.cluster.engflow.com/invocation/9f917b34-099d-49ca-aa75-b4acb6f02fad) on master @ [2859e5fab431536ec77f646787af637eec3d185c](https://github.com/cockroachdb/cockroach/commits/2859e5fab431536ec77f646787af637eec3d185c):


```
=== RUN   TestClusterInflightTraces
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestClusterInflightTraces3123968450
    test_log_scope.go:81: use -show-logs to present logs inline
    collector_test.go:344: -- test log scope end --
test logs left over in: outputs.zip/logTestClusterInflightTraces3123968450
--- FAIL: TestClusterInflightTraces (12.76s)
=== RUN   TestClusterInflightTraces/shared-process
    collector_test.go:339: 
        	Error Trace:	pkg/util/tracing/collector/collector_test_test/pkg/util/tracing/collector/collector_test.go:339
        	Error:      	"map[%!s(int=1):[root root.child root.child.remotechilddone]]" should have 0 item(s), but has 1
        	Test:       	TestClusterInflightTraces/shared-process
    --- FAIL: TestClusterInflightTraces/shared-process (4.22s)
```
<p>Parameters: <code>attempt=1</code>
, <code>run=8</code>
, <code>shard=1</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-inf-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterInflightTraces.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-33851

pkg/util/tracing/collector/collector_test.TestClusterInflightTraces [failed](https://tanzanite.cluster.engflow.com/invocation/8dd72aed-8fea-46dd-8ce5-44c52fa58893) on master @ [caff15394fcbe37208b46b2973714c27cc3a1417](https://github.com/cockroachdb/cockroach/commits/caff15394fcbe37208b46b2973714c27cc3a1417):


```
=== RUN   TestClusterInflightTraces
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestClusterInflightTraces2613737037
    test_log_scope.go:81: use -show-logs to present logs inline
    collector_test.go:344: -- test log scope end --
test logs left over in: outputs.zip/logTestClusterInflightTraces2613737037
--- FAIL: TestClusterInflightTraces (7.57s)
=== RUN   TestClusterInflightTraces/shared-process
    collector_test.go:339: 
        	Error Trace:	pkg/util/tracing/collector/collector_test_test/pkg/util/tracing/collector/collector_test.go:339
        	Error:      	"map[%!s(int=1):[root root.child root.child.remotechilddone]]" should have 0 item(s), but has 1
        	Test:       	TestClusterInflightTraces/shared-process
    --- FAIL: TestClusterInflightTraces/shared-process (2.81s)
```
<p>Parameters: <code>attempt=1</code>
, <code>run=25</code>
, <code>shard=1</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterInflightTraces.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/util/tracing/collector/collector_test.go
+++ b/pkg/util/tracing/collector/collector_test.go
@@ -237,6 +237,7 @@ func TestClusterInflightTraces(t *testing.T) {
 			}
 
 			type testCase struct {
+				name    string
 				servers []serverutils.ApplicationLayerInterface
 				dbs     []*gosql.DB
 				// otherServers, if set, represents the servers corresponding to
@@ -248,6 +249,7 @@ func TestClusterInflightTraces(t *testing.T) {
 			switch config {
 			case "single-tenant":
 				testCases = []testCase{{
+					name:    "system-tenant",
 					servers: []serverutils.ApplicationLayerInterface{tc.Servers[0], tc.Servers[1]},
 					dbs:     systemDBs,
 				}}
@@ -263,11 +265,13 @@ func TestClusterInflightTraces(t *testing.T) {
 				}
 				testCases = []testCase{
 					{
+						name:         "application-tenant",
 						servers:      tenants,
 						dbs:          dbs,
 						otherServers: systemServers,
 					},
 					{
+						name:         "system-tenant",
 						servers:      systemServers,
 						dbs:          systemDBs,
 						otherServers: tenants,
@@ -286,11 +290,13 @@ func TestClusterInflightTraces(t *testing.T) {
 				}
 				testCases = []testCase{
 					{
+						name:         "application-tenant",
 						servers:      tenants,
 						dbs:          dbs,
 						otherServers: systemServers,
 					},
 					{
+						name:         "system-tenant",
 						servers:      systemServers,
 						dbs:          systemDBs,
 						otherServers: tenants,
@@ -299,45 +305,47 @@ func TestClusterInflightTraces(t *testing.T) {
 			}
 
 			for _, tc := range testCases {
-				// Set up the traces we're going to look for.
-				localTraceID, _, cleanup := setupTraces(tc.servers[0].Tracer(), tc.servers[1].Tracer())
-				defer cleanup()
+				t.Run(tc.name, func(t *testing.T) {
+					// Set up the traces we're going to look for.
+					localTraceID, _, cleanup := setupTraces(tc.servers[0].Tracer(), tc.servers[1].Tracer())
+					defer cleanup()
 
-				// Create some other spans on tc.otherServers, that we don't
-				// expect to find.
-				const otherServerSpanName = "other-server-span"
-				for _, s := range tc.otherServers {
-					sp := s.Tracer().StartSpan(otherServerSpanName)
-					defer sp.Finish()
-				}
-
-				// We're going to query the cluster_inflight_traces through
-				// every SQL instance.
-				for _, db := range tc.dbs {
-					rows, err := db.Query(
-						"SELECT node_id, trace_str FROM crdb_internal.cluster_inflight_traces "+
-							"WHERE trace_id=$1 ORDER BY node_id",
-						localTraceID)
-					require.NoError(t, err)
-
-					expSpans := map[int][]string{
-						1: {"root", "root.child", "root.child.remotechilddone"},
-						2: {"root.child.remotechild"},
+					// Create some other spans on tc.otherServers, that we don't
+					// expect to find.
+					const otherServerSpanName = "other-server-span"
+					for _, s := range tc.otherServers {
+						sp := s.Tracer().StartSpan(otherServerSpanName)
+						defer sp.Finish()
 					}
-					for rows.Next() {
-						var nodeID int
-						var trace string
-						require.NoError(t, rows.Scan(&nodeID, &trace))
-						exp, ok := expSpans[nodeID]
-						require.True(t, ok)
-						delete(expSpans, nodeID) // Consume this entry; we'll check that they were all consumed.
-						for _, span := range exp {
-							require.Contains(t, trace, "=== operation:"+span)
+
+					// We're going to query the cluster_inflight_traces through
+					// every SQL instance.
+					for _, db := range tc.dbs {
+						rows, err := db.Query(
+							"SELECT node_id, trace_str FROM crdb_internal.cluster_inflight_traces "+
+								"WHERE trace_id=$1 ORDER BY node_id",
+							localTraceID)
+						require.NoError(t, err)
+
+						expSpans := map[int][]string{
+							1: {"root", "root.child", "root.child.remotechilddone"},
+							2: {"root.child.remotechild"},
 						}
-						require.NotContains(t, trace, "=== operation:"+otherServerSpanName)
+						for rows.Next() {
+							var nodeID int
+							var trace string
+							require.NoError(t, rows.Scan(&nodeID, &trace))
+							exp, ok := expSpans[nodeID]
+							require.True(t, ok)
+							delete(expSpans, nodeID) // Consume this entry; we'll check that they were all consumed.
+							for _, span := range exp {
+								require.Contains(t, trace, "=== operation:"+span)
+							}
+							require.NotContains(t, trace, "=== operation:"+otherServerSpanName)
+						}
+						require.Len(t, expSpans, 0)
 					}
-					require.Len(t, expSpans, 0)
-				}
+				})
 			}
 		})
 	}

